core/utils/utils.go

```diff
@@ -67,6 +68,20 @@ func Uint64ToHex(i uint64) string {
 	return fmt.Sprintf("0x%x", i)
 }
 
+var maxUint256 = common.HexToHash("0x" + strings.Repeat("f", 64)).Big()
+
+// Uint256ToBytes(x) is x represented as the bytes of a uint256
+func Uint256ToBytes(x *big.Int) (uint256 []byte, err error) {
+	if x.Cmp(maxUint256) > 0 {
+		return nil, fmt.Errorf("too large to convert to uint256")
+	}
+	uint256 = common.LeftPadBytes(x.Bytes(), EVMWordByteLen)
+	if x.Cmp(big.NewInt(0).SetBytes(uint256)) != 0 {
+		panic("failed to round-trip uint256 back to source big.Int")
```

Comment:
Let's error instead of panicking. I think the only time it is acceptable to panic is at startup, and even then an error is definitely better.

Reply:
This is an assertion of something which can never happen. Returning an error doesn't make much sense because it would mean that this function or golang itself is broken, and the calling context can't handle that. I think it adds value, though, because in the case that roundtripping ever *does* fail, it fails as early as possible, before the bogus return value gets to corrupt downstream calculations and trigger some much less legible failure.